SplatoonScripts/Duties/Stormblood/UCOB Twisters.cs

```diff
@@ -0,0 +1,125 @@
+﻿using Dalamud.Game.ClientState.Objects.SubKinds;
+using Dalamud.Game.ClientState.Objects.Types;
+using Dalamud.Game.Text;
+using Dalamud.Game.Text.SeStringHandling;
+using Dalamud.Plugin;
+using ECommons;
+using ECommons.Configuration;
+using ECommons.DalamudServices;
+using ECommons.GameFunctions;
+using ECommons.Hooks;
+using ECommons.ImGuiMethods;
+using ECommons.Logging;
+using ECommons.MathHelpers;
+using ImGuiNET;
+using Splatoon;
+using Splatoon.SplatoonScripting;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace SplatoonScriptsOfficial.Duties.Stormblood;
+
+public class UCOB_Twisters : SplatoonScript
+{
+    public override HashSet<uint> ValidTerritories => new() { 733 };
+
+    public override Metadata? Metadata => new(1, "Enthusiastus");    
+
+    private List<Element> _elements = new List<Element>();
+    private bool active = false;
+    private List<IPlayerCharacter> players = FakeParty.Get().ToList();
+
+    Config Conf => this.Controller.GetConfig<Config>();
+
+    string TestOverride = "";
+    IPlayerCharacter PC => TestOverride != "" && FakeParty.Get().FirstOrDefault(x => x.Name.ToString() == TestOverride) is IPlayerCharacter pc ? pc : Svc.ClientState.LocalPlayer!;
+
+    public override void OnSetup()
+    {
+        for(var i = 0; i < 8; i++)
+        {
+            var elem = new Element(0) { Enabled = false, radius = 1.0f, thicc = 2f, color = 0xFF03BAFC };
+            this.Controller.TryRegisterElement($"twister{i}", elem);
+            _elements.Add(elem);
+        }
+    }
+
+    //> [08.09.2024 21:37:07 + 02:00] Message: Twintania starts casting 9898 (1482>9898)
+    public override void OnStartingCast(uint source, uint castId) {
+        if(castId == 9898)
+        {
+            //DuoLog.Debug($"Twistin' time. src {source}");
+            active = true;
+            Task.Delay(2300).ContinueWith(_ =>
```

Comment:
Moving into a different thread is not a good idea here. You can use:
`this.Controller.ScheduleReset(2300);`
instead, and override OnReset method which will be called after 2.3s in this case:
```
public override void OnReset()
{
    Off();
}
```
Additionally, OnReset will be automatically called each time combat starts/ends, party wipes, or script being enabled/disabled, so it can be used as a generic reset to default state.
`this.Controller.Reset();` can be called manually if needed, calling overriden OnReset methos and cancelling any pending reset tasks.

Reply:
If you want, you can try it yourself, otherwise I'll merge and modify it as needed

Reply:
Thank you!
By doing so, you will be running code in a different thread which, unless you do extra precautions, enables situation where two threads (main game thread where Splatoon runs and task thread) can modify data at the same time which causes random unexpected errors.